pkg/proxy/iptables/proxier.go

```diff
@@ -244,52 +170,32 @@ type endpointsInfo struct {
 }
 
 // returns a new proxy.Endpoint which abstracts a endpointsInfo
-func newEndpointsInfo(IP string, port int, isLocal bool) proxy.Endpoint {
-	return &endpointsInfo{
-		endpoint: net.JoinHostPort(IP, strconv.Itoa(port)),
-		isLocal:  isLocal,
-	}
+func customizeEndpointInfo(IP string, port int, isLocal bool, infoCommon *proxy.EndpointInfoCommon) proxy.Endpoint {
+	return &endpointsInfo{EndpointInfoCommon: infoCommon}
```

Comment:
> if we assume infoCommon has these values populated, no need to accept these as function parameters

Yeah the assumption is infoCommon should have been populated because `newEndpointInfoCommon()` is always called first. My thought here was to provide an interface with more flexibility --- the other args are not used just because the iptables proxier doesn't need them, but it may not be true for all proxiers (in future). But anyway it seems better to just remove those unused args now. Fixed.